src/lava/magma/runtime/runtime.py

```diff
@@ -201,56 +205,97 @@ def start(self, run_condition: AbstractRunCondition):
         else:
             print("Runtime not initialized yet.")
 
-    def _run(self, run_condition):
+    def _get_resp_for_run(self):
+        """
+        Gets response from RuntimeServices
+        """
+        if self._is_running:
+            for recv_port in self.service_to_runtime:
+                data = recv_port.recv()
+                if enum_equal(data, MGMT_RESPONSE.REQ_PAUSE):
+                    self._req_paused = True
+                elif enum_equal(data, MGMT_RESPONSE.REQ_STOP):
+                    self._req_stop = True
+                elif not enum_equal(data, MGMT_RESPONSE.DONE):
+                    if enum_equal(data, MGMT_RESPONSE.ERROR):
+                        # Receive all errors from the ProcessModels
+                        error_cnt = 0
+                        for actors in \
+                                self._messaging_infrastructure.actors:
+                            actors.join()
+                            if actors.exception:
+                                _, traceback = actors.exception
+                                print(traceback)
+                                error_cnt += 1
+                        self._error = True
+                    else:
+                        raise RuntimeError(f"Runtime Received {data}")
+            if self._req_paused:
+                self._req_paused = False
+                self.pause()
```

Comment:
It might actually make sense to print an info message to let user know why execution suddenly halts when some sub process requests a PAUSE or STOP.
XYZ process requested a pause, ...